pkg/apis/v1alpha1/common.go

```diff
@@ -151,3 +158,35 @@ func GetAPITemplate(templateKind string) (client.Object, error) {
 	}
 	return template, nil
 }
+
+type TemplateOption struct {
+	// Name of the template to apply
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Selector is a field query over a workload resource.
+	Selector Selector `json:"selector"`
+}
+
+type Selector struct {
+	// MatchFields is a list of field selector requirements. The requirements are ANDed.
+	// +kubebuilder:validation:MinItems=1
+	MatchFields []FieldSelectorRequirement `json:"matchFields"`
+}
+
+type FieldSelectorRequirement struct {
+	// Key is the JSON path in the workload to match against.
+	// e.g. "workload.spec.source.git.url"
```

Comment:
now that it's under `common`, I wonder if we should make this conditional, something like

```
e.g., for workload: `workload.spec...`
e.g., for deliverable: `deliverable.spec..` 
```

so that we catch both cases 🤔